### ISSUE TITLE
MapColumn & ListColumn - Intellij code inspection reports compile error

### ISSUE THREAD

**Issue by gabbah** [outworkers/phantom#654]
Intellij code inspection reports an error for a simple use case, but it compiles and runs just fine. Is the fault with intellij or with the phantom code?
Perhaps some type hint is needed somewhere, to make Intellij's parser do the right thing?

```scala
case class MyCaseClass()
class Table extends CassandraTable[Table, MyCaseClass] {
  object id extends StringColumn(this) with PartitionKey
  object mapCol extends MapColumn[UUID, String](this) 
/* Intellij marks "this" as error and expects table: CassandraTable[UUID, String] */
}
```

The same problem is there with ListColumn as well, but not for simple column types.

**Comment by alexflav23** [comment]
@gabbah

We are aware of this, it's because those columns are nothing more than type aliases and InteliJ is not smart enough to dealias a type and see if the constructor of that type takes params. We are fixing this in the next minor increment, namely 2.7.0, where the table schema will no longer require passing in the table as an argument to columns.

```
case class MyCaseClass()
class Table extends CassandraTable[Table, MyCaseClass] {
  object id extends StringColumn with PartitionKey
  object mapCol extends MapColumn[UUID, String] 
/* Intellij marks "this" as error and expects table: CassandraTable[UUID, String] */
}
```

**Comment by alexflav23** [comment]
Hi @gabbah 

I am going to close this ticket now since it's really a highlighting problem in InteliJ and it's gotten fixed anyway, 2.7.0 is going to be ready by end of next week and you can benefit from the fixes.

Regards,
